Sources/Sentry/SentryAppStartTracker.m

```diff
@@ -0,0 +1,181 @@
+#import "SentryAppStartMeasurement.h"
+#import "SentryAppStateManager.h"
+#import "SentryLog.h"
+#import <Foundation/Foundation.h>
+#import <SentryAppStartTracker.h>
+#import <SentryAppState.h>
+#import <SentryClient+Private.h>
+#import <SentryCurrentDateProvider.h>
+#import <SentryDispatchQueueWrapper.h>
+#import <SentryFileManager.h>
+#import <SentryHub.h>
+#import <SentryInternalNotificationNames.h>
+#import <SentryLog.h>
+#import <SentrySDK+Private.h>
+#import <SentrySpan.h>
+#import <SentrySysctl.h>
+
+#if SENTRY_HAS_UIKIT
+#    import <UIKit/UIKit.h>
+#endif
+
+static NSDate *appStart = nil;
+
+@interface
+SentryAppStartTracker ()
+
+@property (nonatomic, strong) SentryOptions *options;
+@property (nonatomic, strong) id<SentryCurrentDateProvider> currentDate;
+@property (nonatomic, strong) SentryAppState *previousAppState;
+@property (nonatomic, strong) SentryDispatchQueueWrapper *dispatchQueue;
+@property (nonatomic, strong) SentryAppStateManager *appStateManager;
+@property (nonatomic, strong) SentrySysctl *sysctl;
+@property (nonatomic, assign) BOOL wasInBackground;
+
+@end
+
+@implementation SentryAppStartTracker
+
++ (void)load
+{
+    // Invoked whenever this class is added to the Objective-C runtime. That's the best
+    // approximation of the app start time we can get.
+    appStart = [NSDate date];
+}
+
+- (instancetype)initWithOptions:(SentryOptions *)options
+            currentDateProvider:(id<SentryCurrentDateProvider>)currentDateProvider
+           dispatchQueueWrapper:(SentryDispatchQueueWrapper *)dispatchQueueWrapper
+                appStateManager:(SentryAppStateManager *)appStateManager
+                         sysctl:(SentrySysctl *)sysctl
+{
+    if (self = [super init]) {
+        self.options = options;
+        self.currentDate = currentDateProvider;
+        self.dispatchQueue = dispatchQueueWrapper;
+        self.appStateManager = appStateManager;
+        self.sysctl = sysctl;
+#if SENTRY_HAS_UIKIT
+        self.previousAppState = [self.appStateManager loadCurrentAppState];
+#endif
+        self.wasInBackground = NO;
+    }
+    return self;
+}
+
+- (void)start
+{
+
+#if SENTRY_HAS_UIKIT
```

Comment:
Does it make sense to have this class if we're not on UIKit? Most of the logic is guarded by having it.

Reply:
I changed it. Good point.